### PR TITLE
Remove status from transverse

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: BackofficeBundle/Manager/NodeManager.php [hunk 1]
```diff
@@ -92,6 +97,20 @@ public function createNewLanguageNode(NodeInterface $node, $language)
     }
 
     /**
+     * @param NodeInterface $node
+     *
+     * @return StatusInterface
+     */
+    protected function getInitialStatus(NodeInterface $node)
+    {
+        if ($node->getNodeType() == NodeInterface::TYPE_GENERAL) {
```
Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: Why do you make a different test here and in the transformer : 
  - One for only node transverse
  - One for all general node

What about the other nodes?

File: BackofficeBundle/Tests/Manager/NodeManagerTest.php [hunk 2]
```diff
@@ -226,7 +228,12 @@ public function generateConsistencyNode()
...
     {
-        $node = $this->manager->initializeNewNode();
+        $son = Phake::mock('OpenOrchestra\ModelInterface\Model\NodeInterface');
+        Phake::when($son)->getNodeType()->thenReturn('fakeType');
+        $fakeParentId = 'fakeParentId';
+        Phake::when($this->nodeRepository)->findOneByNodeIdAndLanguageAndVersionAndSiteId(Phake::anyParameters())->thenReturn($son);
+
+        $node = $this->manager->initializeNewNode($fakeParentId);
 
         $this->assertInstanceOf($this->nodeClass, $node);
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/Manager/NodeManagerTest.php]: You could also add a check on the nodeType of the node returned!

**Comment by nicolasThal** [comment]
I think that you should also add a test for the difference between the node transverse and the general nodes

File: BackofficeBundle/Manager/NodeManager.php (hunk 1 above)
Comment (itkg-nanne) [inline BackofficeBundle/Manager/NodeManager.php]: In fact TYPE_GENERAL is transverse. Others are TYPE_DEFAULT...

Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: Yes, but they can be TYPE_GENERAL nodes which are not transverse node.

The point of my remark was not about that but about the different test you where doing in the nodeTransformer and at this line

Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/718/files#diff-51685802d3e8eddde0376c8bb9d9ae88R134 on this line

File: BackofficeBundle/Manager/NodeManager.php [hunk 3]
```diff
@@ -92,6 +97,20 @@ public function createNewLanguageNode(NodeInterface $node, $language)
...
+            return $this->statusRepository->findOneByEditable();
```
Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: this method should not be used, use this method instead findOneByInitial

Comment (itkg-nanne) [inline BackofficeBundle/Manager/NodeManager.php]: We're not sure that initial is not published on true. If transverse is initial and initial is published true thn we can't edit.

**Comment by itkg-nanne** [comment]
c'est fait @nicolasThal

File: BackofficeBundle/Manager/NodeManager.php [hunk 4]
```diff
@@ -92,6 +97,20 @@ public function createNewLanguageNode(NodeInterface $node, $language)
     }
 
     /**
+     * @param NodeInterface $node
+     *
+     * @return StatusInterface
+     */
+    protected function getInitialStatus(NodeInterface $node)
```
Comment (nicolasThal) [inline BackofficeBundle/Manager/NodeManager.php]: I haven't seen it yet but this method is not consistent.

You cannot say getInitialStatus and then return only an editable status.

Comment (itkg-nanne) [inline BackofficeBundle/Manager/NodeManager.php]: I use this beacause I need the initial status of the node, not the status which is flagged initial.
But I change it. @nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**